generate/templates/manual/include/worker_pool.h

```diff
@@ -58,9 +58,13 @@ class WorkerPool {
   WorkerPool& operator=(const WorkerPool &other) = delete;
   WorkerPool& operator=(WorkerPool &&other) = delete;
 
-  bool Init(std::vector< std::shared_ptr<Worker> > workers);
-  bool InsertWork(std::unique_ptr<WorkItem> &&work);
+  void Init(std::vector< std::shared_ptr<Worker> > workers);
+  void InsertWork(std::unique_ptr<WorkItem> &&work);
   void Shutdown();
+  bool Status() { return m_status.workersInitialize && m_status.workersExecute && !m_status.shutdownEarly; }
+  bool StatusWorkersInitialize() { return m_status.workersInitialize; }
+  bool StatusWorkersExecute() { return m_status.workersExecute; }
+  bool StatusShutdownEarly() { return !m_status.shutdownEarly; }
```

Comment:
Mutex is required to read, not just write. Compiler optimizations can prevent these shared values from ever being read. See https://devblogs.microsoft.com/oldnewthing/20180924-00/?p=99805.

I would instead leverage `std::atomic<bool>` instead of leveraging any mutexes here at all.

Reply:
Good catch! thank you!

In the fix I've reduced the status to just one atomic enumeration variable